pkg/migration/templates/kubeadm_controlplane_azure.yaml.tmpl

```diff
@@ -0,0 +1,286 @@
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+kind: KubeadmControlPlane
+metadata:
+  name: {{.ClusterID}}-control-plane
+  namespace: default
+  annotations:
+    controlplane.cluster.x-k8s.io/skip-coredns: "true"
+spec:
+  replicas: 1
+  version: {{ .K8sVersion }}
+  infrastructureTemplate:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
+    kind: AzureMachineTemplate
+    name: {{.ClusterID}}-control-plane
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      apiServer:
+        certSANs:
+          - api.{{.ClusterID}}.k8s.{{.InstallationBaseDomain}}
+        extraArgs:
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: azure
+          etcd-prefix: giantswarm.io
+          encryption-provider-config: "/etc/kubernetes/encryption/k8s-encryption-config.yaml"
+        extraVolumes:
+        - hostPath: /etc/kubernetes/azure.json
+          mountPath: /etc/kubernetes/azure.json
+          name: cloud-config
+          readOnly: true
+        - hostPath: /etc/kubernetes/encryption/
+          mountPath: /etc/kubernetes/encryption/
+          name: encryption
+          readOnly: true
+        timeoutForControlPlane: 20m
+      controllerManager:
+        extraArgs:
+          allocate-node-cidrs: "true"
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: azure
+          cluster-name: {{.ClusterID}}
+        extraVolumes:
+        - hostPath: /etc/kubernetes/azure.json
+          mountPath: /etc/kubernetes/azure.json
+          name: cloud-config
+          readOnly: true
+      controlPlaneEndpoint: api.{{.ClusterID}}.k8s.{{.InstallationBaseDomain}}:443
+      etcd:
+        local:
+          dataDir: /var/lib/etcddisk/etcd
+          extraArgs:
+            "initial-cluster-state": existing
+            "initial-cluster": "$ETCD_INITIAL_CLUSTER"
+          imageTag: {{ .EtcdVersion }}
+          imageRepository: "quay.io/giantswarm"
+      networking:
+        dnsDomain: cluster.local
+        serviceSubnet: 172.31.0.0/16
+    diskSetup:
+      filesystems:
+      - device: /dev/disk/azure/scsi1/lun0
+        extraOpts:
+        - -E
+        - lazy_itable_init=1,lazy_journal_init=1
+        filesystem: ext4
+        label: etcd_disk
+      - device: ephemeral0.1
+        filesystem: ext4
+        label: ephemeral0
+        replaceFS: ntfs
+      partitions:
+      - device: /dev/disk/azure/scsi1/lun0
+        layout: true
+        overwrite: false
+        tableType: gpt
+    files:
+    - contentFrom:
+        secret:
+          key: control-plane-azure.json
+          name: {{.ClusterID}}-control-plane-azure-json
+      owner: root:root
+      path: /etc/kubernetes/azure.json
```

Comment:
We also have that defined as host volume. Is that intended?

Reply:
Ping @whites11 ?

This is direct copypaste from your test YAML.

Reply:
this part comes straight out of `clusterctl config`.
Not sure what you mean @kopiczko with `defined as host volume`

Reply:
I mean this https://github.com/giantswarm/capi-migration/pull/24/files#diff-903034c1a10c2a1b1bfd4177147376690795feb0b3848b14e8c0b73729dfd6f2R26-R29 - possibly I don't understand something.

Reply:
So, lines 78-81 create the file in the server, while lines 26-29 are added to the apiserver static manifest to mount the file inside the container